OptimoveSDK/Sources/Classes/Optimobile/AnalyticsHelper.swift

```diff
@@ -35,12 +37,49 @@ class AnalyticsHelper {
             }
             self.syncEvents(context: self.analyticsContext)
         }
+
+        finishedInitializationToken = NotificationCenter.default
+            .addObserver(forName: .optimobileInializationFinished, object: nil, queue: nil) { [weak self] notification in
+                guard let self = self else { return }
+                if let analyticsContext = self.analyticsContext {
+                    self.syncEvents(context: analyticsContext)
+                }
+                self.maybeSendPendingMetrics()
+                Logger.debug("Notification \(notification.name.rawValue) was processed")
+            }
     }
 
     deinit {
         eventsHttpClient.invalidateSessionCancellingTasks(false)
     }
 
+    private func maybeSendPendingMetrics() {
+        let removeAllPendingMetrics = {
+            do {
+                try self.pendingAnalytics.removeAll()
+            } catch {
+                Logger.error("Failed to remove all pending metrics: " + error.localizedDescription)
+            }
+        }
+        do {
+            let metrics = try pendingAnalytics.readAll()
+            Logger.debug("Found \(metrics.count) pending metrics")
+            metrics.forEach { metric in
+                self.trackEvent(
+                    eventType: metric.eventType.rawValue,
+                    atTime: Date(timeIntervalSince1970: Double(metric.timestamp)),
+                    properties: metric.properties,
+                    immediateFlush: true,
+                    onSyncComplete: { _ in
+                        removeAllPendingMetrics()
```

Comment:
we track event for each metric, but removeAll after each one completes. it's probably minor and not important enough but theoretically could miss events if
- 1st succeeds, rest fail
- receive new event while syncing these

Reply:
You right. Fixed

Reply:
could there be thread safety issues?  when we remove 1 we read all and set all. 

however, since delivered events are tech debt, let's deal later